src/Microsoft.AspNetCore.Routing/Matchers/LinearSearchJumpTable.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    internal class LinearSearchJumpTable : JumpTable
+    {
+        private readonly int _default;
+        private readonly int _exit;
+        private readonly (string text, int destination)[] _entries;
+
+        public LinearSearchJumpTable(int @default, int exit, (string text, int destination)[] entries)
```

Comment:
@JamesNK do you still think it's unclear that these all refer to state indices? what about names like `defaultDestination`, `exitDestination`, etc? Better enough to be longer?

Reply:
I like Destination suffix.

What convinced me was `int segment` in some places and `PathSegment segment` in others.

Reply:
Ok, will do then.